### PR TITLE
Enable https://github.com/opentripplanner/otp-react-redux/pull/1266 in call taker

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
**Description:**
Enables `syncSortWithDepartArrive` to work in call taker. Mostly the same code

**PR Checklist:**
- [ ] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [ ] Are all languages supported (Internationalization/Localization)?
- [ ] Are appropriate Typescript types implemented?

### CODE REVIEW

File: lib/components/form/call-taker/date-time-picker.tsx [hunk 1]
```diff
@@ -187,6 +197,18 @@ const DateTimeOptions = ({
         })
       })
     }
+
+    if (
+      syncSortWithDepartArrive &&
+      DepartArriveTypeMap[departArrive as DepartArriveValue] !== sort.type
```
Comment (daniel-heppner-ibigroup) [inline lib/components/form/call-taker/date-time-picker.tsx]: cast `departArrive` wherever you get it from

File: lib/components/form/call-taker/date-time-picker.tsx [hunk 2]
```diff
@@ -283,11 +305,18 @@ const DateTimeOptions = ({
 
 // connect to the redux store
 const mapStateToProps = (state: any) => {
```
Comment (daniel-heppner-ibigroup) [inline lib/components/form/call-taker/date-time-picker.tsx]: can you make this type AppState

**Review by daniel-heppner-ibigroup — CHANGES_REQUESTED** [review]
fix type

File: lib/components/form/call-taker/date-time-picker.tsx (hunk 1 above)
Comment (miles-grant-ibigroup) [reply lib/components/form/call-taker/date-time-picker.tsx]: It's a `useState`. TS doesn't like me typing it at definition time

**Comment by miles-grant-ibigroup** [comment]
@daniel-heppner-ibigroup Typescript is giving me some trouble

**Review by daniel-heppner-ibigroup — APPROVED** [review]
I fixed it for you please look at my commit

**Comment by miles-grant-ibigroup** [comment]
Much cleaner thanks

**Review by josh-willis-arcadis — APPROVED** [review]
LGTM!

**Pull request merged**